### DISCUSSION TITLE
blog/github-actions-deploy-dotnet-webapi-to-amazon-ecs/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#52] [General]
# blog/github-actions-deploy-dotnet-webapi-to-amazon-ecs/

Let’s automate the deployment of a brand-new .NET 9 Web API to Amazon ECS using a GitHub Actions CI/CD pipeline. In this guide, we’ll walk through building a .NET 9 Docker image directly on GitHub, pushing it to Amazon Elastic Container Registry (ECR), creating an ECS task definition, and configuring an ECS service to fetch and deploy the latest Docker image. By the end, your application will be up and running on ECS with a fully automated workflow.

https://codewithmukesh.com/blog/github-actions-deploy-dotnet-webapi-to-amazon-ecs/

**Comment by sureshramasundaram** [comment]
Thanks for the nice article.

**Comment by jakenuts** [comment]
That's easily the most succinct yet perfectly detailed walkthrough of ECS deployment I've ever read. One thing to note (I learned this yesterday) is that by default timing ECS will run 5 "health checks" every 30 seconds before switching traffic to your new image, so a 2min 30sec delay. For deployment one or two checks spaced 5-10 seconds apart would probably be enough so that's 20 seconds to live.

**Comment by pawanprateek16** [comment]
What a Great Explanation Mukesh. I love your article and .NET Technology material Thanks for Sharing Great work keep going

**Comment by timsar2** [comment]
Hi,

I just wanted to say that your articles are excellent. I really enjoy reading them and I’ve learned a lot from the way you explain concepts so clearly.

I was curious about your opinion on something: with the upcoming .NET 11 release, do you think the gap between .NET and Go will become smaller when the goal is just building minimal Web APIs?

And if not, do you have any plans to update or expand your Go content in the future, similar to the course you created for .NET?

Thanks again for the great content.
